mayday/file_test.go

```diff
@@ -0,0 +1,35 @@
+package mayday_test
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/coreos/mayday/mayday"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNonexistentFile(t *testing.T) {
+	file := mayday.File{Path: "/etc/nonexistent"}
+	err := file.Collect(workspace)
+	assert.Equal(t, err.Error(), `error opening source file: open /etc/nonexistent: no such file or directory`)
+}
+
+func TestFile(t *testing.T) {
+	file := mayday.File{Path: "/proc/meminfo"}
+	res := file.Collect(workspace)
+	assert.Nil(t, res)
+
+	f, err := ioutil.ReadFile(workspace + "/proc/meminfo")
+	assert.Nil(t, err)
+	assert.Contains(t, string(f), "MemTotal")
+}
+
+func TestFileWithLink(t *testing.T) {
+	file := mayday.File{Path: "/etc/crontab", Link: "cronfig"}
```

Comment:
Given that this is a unit test, it should support at least darwin and linux and not depend on external state wherever possible.

I would lean towards refactoring `mayday.File.Path` to be an `io.ReadCloser` interface type - that way you can effectively mock up an ingest file to mayday without depending on filesystem state- which is considered bad practice in the context of unit tests.